cli/script/command-executor.ts

```diff
@@ -1020,6 +1057,50 @@ function getPackageMetricsString(packageObject: PackageWithMetrics): string {
     return returnString;
 }
 
+function getReactNativeProjectAppVersion(platform: string, projectName: string): Promise<string> {
+    if (platform === "ios") {
+        try {
+            var infoPlistContents: string = fs.readFileSync(path.join("iOS", projectName, "Info.plist")).toString();
+        } catch (err) {
+            try {
+                infoPlistContents = fs.readFileSync(path.join("iOS", "Info.plist")).toString();
+            } catch (err) {
+                throw new Error(`Unable to find or read "Info.plist" in the "iOS/${projectName}" or "iOS" folders.`);
+            }
+        }
+        
+        var appVersionRegex: RegExp = /CFBundleShortVersionString\s*<\s*\/key\s*>\s*<\s*string\s*>(.*?)<\s*\/string\s*>/;
+        var parsedInfoPlist: string[] = infoPlistContents.match(appVersionRegex);
+        if (parsedInfoPlist && parsedInfoPlist[1]) {
+             if (semver.valid(parsedInfoPlist[1]) === null) {
+                throw new Error("Please update \"Info.plist\" to use a semver compliant \"CFBundleShortVersionString\", for example \"1.0.3\".");
+            } else {
+                return Q(parsedInfoPlist[1]);
+            }
+        } else {
+            throw new Error("Unable to parse the \"CFBundleShortVersionString\" from \"Info.plist\".");
+        }
+    } else {
+        try {
+            var buildGradleContents: string = fs.readFileSync(path.join("android", "app", "build.gradle")).toString();
+        } catch (err) {
+            throw new Error("Unable to find or read \"build.gradle\" in the \"android/app\" folder.");
+        }
+        
+        var appVersionRegex: RegExp = /versionName\s+"(.*?)"/;
```

Comment:
If we parsed this in a smarter way, e.g. make sure we are looking at android.defaultConfig.versionName, would it avoid the scenario where other plugin or config objects have the versionName method?

Granted, that wouldn't handle the case where it is a dynamic value, e.g `getVersionName()`. If its a rare enough case, I guess its okay to allow those users to fall back to the original way of manually specifying the appVersion to our CLI?